### DISCUSSION TITLE
Tailwind + TypeScript: Unsafe assignment of type any[] in tailwind.config.ts

### DISCUSSION THREAD

**Discussion by Stadly** [tailwindlabs/tailwindcss#12578] [Help] [answered]
Consider the following `tailwind.config.ts` file:
```ts
import type { Config } from "tailwindcss";

export default {
  content: [],
  theme: {
    extend: {},
  },
  plugins: [require("@tailwindcss/forms")],
} satisfies Config;
```

ESLint complains about `plugins: [require("@tailwindcss/forms")]`:

```
error  Unsafe assignment of type any[] to a variable of type (PluginCreator | { handler: PluginCreator; config?: Partial<Config>; } | { (options: any): { handler: PluginCreator; config?: Partial<...>; }; __isOptionsFunction: true; } | undefined)[]  @typescript-eslint/no-unsafe-assignment
```

This error surfaced when updating from TypeScript 5.2.2 to 5.3.2. How can I fix it?

I've created a reproduction on StackBlitz: https://stackblitz.com/edit/stackblitz-starters-zre7x7?file=tailwind.config.ts
Just run `npx eslint .` to see the error.

**Comment by makibat** [comment]
The plugin returns any. It's trying to assign type `any[]` to the property `plugins` which expects a type defined in the `Config` which cannot be `any[]`. The eslint rule is correct and works as expected. 

I'm not sure what changed regarding the type-checking behaviour between versions 5.2 and 5.3, so I'd suggest one of these two options:

Either downgrade to version 5.2, or if you absolutely need 5.3, add `/* eslint-disable @typescript-eslint/no-unsafe-assignment */` to your file until further updates.

**Comment by Stadly** [comment, marked as answer]
I was able to fix it using `import` instead of `require`:

```ts
import type { Config } from "tailwindcss";
import * as forms from "@tailwindcss/forms";

export default {
  content: [],
  theme: {
    extend: {},
  },
  plugins: [forms],
} satisfies Config;
```
